docs/content/concepts/assets/graph-backed-assets.mdx

```diff
@@ -0,0 +1,208 @@
+
+
+@op
+def store_files(files):
+    return files.to_sql(name="slack_files", con=create_db_connection())
+
+
+@graph
+def store_slack_files_in_sql():
+    return store_files(fetch_files_from_slack())
+
+
+graph_asset = AssetsDefinition.from_graph(store_slack_files_in_sql)
+```
+
+### Defining basic dependencies for graph-backed assets
+
+The `from_graph` attribute on the `AssetsDefinition` object infers upstream and downstream asset dependencies from the graph definition provided. In the most simple case when the graph returns a singular output, Dagster infers the name of the graph to be the outputted asset key.
+
+In the example below, Dagster creates an asset with key `middle_asset` from the `middle_asset` graph. Just like assets defined via <PyObject object="asset" decorator />, each argument to the decorated graph function is an upstream asset name. `middle_asset` depends on `upstream_asset`, and `downstream_asset` depends on `middle_asset`:
+
+```python file=/concepts/assets/graph_backed_asset.py startafter=start_basic_dependencies endbefore=end_basic_dependencies
+from dagster import AssetsDefinition, asset, graph
+
+
+@asset
+def upstream_asset():
+    return 1
+
+
+@graph
+def middle_asset(upstream_asset):
+    return add_one(upstream_asset)
+
+
+middle_asset = AssetsDefinition.from_graph(middle_asset)
+
+
+@asset
+def downstream_asset(middle_asset):
+    return middle_asset + 1
+```
+
+When your graph returns multiple outputs, Dagster infers each output name to be the outputted asset key. In the below example, `two_assets_graph` accepts `upstream_asset` and outputs two assets, `first_asset` and `second_asset`:
+
+```python file=/concepts/assets/graph_backed_asset.py startafter=start_basic_dependencies_2 endbefore=end_basic_dependencies_2
+from dagster import AssetsDefinition, GraphOut, graph
+
+
+@graph(out={"first_asset": GraphOut(), "second_asset": GraphOut()})
+def two_assets_graph(upstream_asset):
+    one, two = two_outputs(upstream_asset)
+    return {"first_asset": one, "second_asset": two}
+
+
+two_assets = AssetsDefinition.from_graph(two_assets_graph)
+```
+
+### Defining explicit dependencies for graph-backed assets
+
+You can also define dependencies for graph-backed assets explicitly via the `asset_keys_by_input_name` and `asset_keys_by_output_name` arguments to `from_graph`:
+
+```python file=/concepts/assets/graph_backed_asset.py startafter=start_explicit_dependencies endbefore=end_explicit_dependencies
+from dagster import AssetsDefinition, GraphOut, graph
+
+
+@graph(out={"one": GraphOut(), "two": GraphOut()})
+def return_one_and_two(zero):
+    one, two = two_outputs(zero)
+    return {"one": one, "two": two}
+
+
+explicit_deps_asset = AssetsDefinition.from_graph(
+    return_one_and_two,
+    keys_by_input_name={"zero": AssetKey("upstream_asset")},
+    keys_by_output_name={
+        "one": AssetKey("asset_one"),
+        "two": AssetKey("asset_two"),
+    },
+)
+```
+
+### Advanced: Subsetting graph-backed assets
```

Comment:
What do we hope including `Advanced` here will achieve?

Reply:
I think the content here is geared toward power users--it requires deeper preexisting knowledge about how ops are combined into graphs and the edge cases there.

My hope is that the "advanced" tag will signal to users (especially beginners) that they can skip this content unless if they are specifically looking for it and prepared to understand the nuances of this feature